src/components/document/document.tsx

```diff
@@ -89,11 +89,13 @@ const CopyButton = ({ onClick }: { onClick: () => void }) => {
   );
 };
 
-const DeleteButton = ({ onClick }: { onClick: () => void }) => {
-  return (
-    <IconButton icon="delete" key="delete" className="action icon-delete"
-                onClickButton={onClick} />
-  );
+const DeleteButton = ({ onClick, enabled }: { onClick: () => void, enabled: boolean }) => {
+    const visibility = enabled ? "enabled" : "disabled";
+    return (
+      <IconButton icon="delete" key={`delete-${visibility}`} className={`action icon-delete delete-${visibility}`}
+                  enabled={enabled} innerClassName={`visibility ${visibility}`}
```

Comment:
```suggestion
                  enabled={enabled} innerClassName={visibility}
```
It seems like we shouldn't need multiple classes here (and as mentioned above, "visibility" isn't really the right word anyway.

Reply:
I was going after the same convention that was already present for the `Share` button but this also makes sense too.